### ISSUE TITLE
Add support for Public IP Address Prefix and additional missing arguments for interfaces on VMSS Scale Set resource in `vmss` sub-module

### ISSUE THREAD

**Issue by jinkang23** [PaloAltoNetworks/terraform-azurerm-swfw-modules#57]
### Is your feature request related to a problem?

This request is related to [Issue #55](https://github.com/PaloAltoNetworks/terraform-azurerm-swfw-modules/issues/55). 

In `vmss` sub-module within `azurerm_linux_virtual_machine_scale_set` resource, following arguments are missing from the `public_ip_addresses` dynamic block:  
- `idle_timeout_in_minutes`
- `public_ip_prefix_id`
- `version`
- `ip_tag`

These arguments, especially `public_ip_prefix_id` are necessary in cases where we want to either 1) BYO-IP with Custom IP Prefix, or 2) reserve a static range of Azure public IPs. In our case, we have onboarded a /24 range via BYO-IP with Custom IP Prefix and have created a smaller /28 range as Azure Public IP prefix. We want to be able to have the Public IP interfaces of VMSS ScaleSet instances to be allocated from this Azure Public IP Prefix instead. 

### Describe the solution you'd like

Here's an example of adding missing arguments to `azurerm_linux_virtual_machine_scale_set` resource:  

```hcl

resource "azurerm_linux_virtual_machine_scale_set" "this" {
  name                 = var.name
  computer_name_prefix = null

   ...<omitted>...

  dynamic "network_interface" {
    for_each = var.interfaces
    iterator = nic

    content {
      name                          = nic.value.name
      primary                       = nic.key == 0 ? true : false

      ...<omitted>...

      ip_configuration {
        name                                         = "primary"
        primary                                      = true
        
       ...<omitted>...

        dynamic "public_ip_address" {
          for_each = nic.value.create_public_ip ? [1] : []
          iterator = pip

          content {
            name              = nic.value.name
            domain_name_label = nic.value.pip_domain_name_label


            idle_timeout_in_minutes = nic.value.idle_timeout_in_minutes
            public_ip_prefix_id = nic.value.public_ip_prefix_id
            version             = nic.value.version

            dynamic "ip_tag" {
              for_each = nic.value.ip_tags

              content {
                tag  = ip_tag.value.tag
                type = ip_tag.value.type
              }
            }
          }
        }
      }
    }
  }
  ```

**Comment by acelebanski** [comment]
Hello @jinkang23, thanks for creating this feature request! We're probably not going to support version other than **IPv4** at the moment. We may introduce IPv6 to all our modules in the future but this would be a wider initiative. Regarding `ip_tag` property, what value would it bring in your opinion? 

For `public_ip_prefix_id`, that's something we could definitely add.

**Comment by jinkang23** [comment]
> Hello @jinkang23, thanks for creating this feature request! We're probably not going to support version other than **IPv4** at the moment. We may introduce IPv6 to all our modules in the future but this would be a wider initiative. Regarding `ip_tag` property, what value would it bring in your opinion?
> 
> For `public_ip_prefix_id`, that's something we could definitely add.

Hi @acelebanski  - I appreciate working this feature request! We currently only have a need for `public_ip_prefix_id`. I only suggested adding the remaining arguments for possible future use. At the moment, I don't have any need for for IPv6 or IP Tags.

**Comment by acelebanski** [comment]
Thanks for explaining @jinkang23. I implemented the improvements in PR #65. Should be released soon when approved.

**Comment by jinkang23** [comment]
> Thanks for explaining @jinkang23. I implemented the improvements in PR #65. Should be released soon when approved.

Perfect, thank you @acelebanski !

**Comment by acelebanski** [comment]
Hello @jinkang23, this functionality will also be introduced in PR #80.
